vars/microserviceBuilderPipeline.groovy

```diff
@@ -47,39 +50,38 @@ def call(body) {
   def build = (config.build ?: env.BUILD ?: "true").toBoolean()
   def deploy = (config.deploy ?: env.DEPLOY ?: "true").toBoolean()
-  def namespace = (config.namespace ?: env.NAMESPACE ?: "").trim()
-  def serviceAccountName = (env.SERVICE_ACCOUNT_NAME ?: "default").trim()
-
-  // these options were all added later. Helm chart may not have the associated properties set.
+  def mvnCommands = (config.mvnCommands == null) ? 'package' : config.mvnCommands
   def test = (config.test ?: (env.TEST ?: "false").trim()).toLowerCase() == 'true'
   def debug = (config.debug ?: (env.DEBUG ?: "false").trim()).toLowerCase() == 'true'
-  def helmSecret = (env.HELM_SECRET ?: "").trim()
-  // will need to check later if user provided chartFolder location
   def userSpecifiedChartFolder = config.chartFolder
   def chartFolder = userSpecifiedChartFolder ?: ((env.CHART_FOLDER ?: "").trim() ?: 'chart')
-  def manifestFolder = config.manifestFolder ?: ((env.MANIFEST_FOLDER ?: "").trim() ?: 'manifests')
-  def libertyLicenseJarBaseUrl = (env.LIBERTY_LICENSE_JAR_BASE_URL ?: "").trim()
   def libertyLicenseJarName = config.libertyLicenseJarName ?: (env.LIBERTY_LICENSE_JAR_NAME ?: "").trim()
-  def alwaysPullImage = (env.ALWAYS_PULL_IMAGE == null) ? true : env.ALWAYS_PULL_IMAGE.toBoolean()
+
+  // Internal 
+  def registry = (env.REGISTRY ?: "").trim()
+  if (registry && !registry.endsWith('/')) registry = "${registry}/"
+  def registrySecret = (env.REGISTRY_SECRET ?: "").trim()
+  def serviceAccountName = (env.SERVICE_ACCOUNT_NAME ?: "default").trim()
+  def mcReleaseName = (env.RELEASE_NAME).toUpperCase()
+  def namespace = (config.namespace ?: env.NAMESPACE ?: "").trim()
+  def helmSecret = (env.HELM_SECRET ?: "").trim()
+  def libertyLicenseJarBaseUrl = (env.LIBERTY_LICENSE_JAR_BASE_URL ?: "").trim()
   def mavenSettingsConfigMap = env.MAVEN_SETTINGS_CONFIG_MAP?.trim()
+  def alwaysPullImage = (env.ALWAYS_PULL_IMAGE == null) ? true : env.ALWAYS_PULL_IMAGE.toBoolean()
   def helmTlsOptions = " --tls --tls-ca-cert=/msb_helm_sec/ca.pem --tls-cert=/msb_helm_sec/cert.pem --tls-key=/msb_helm_sec/key.pem " 
-  def mcReleaseName = (env.RELEASE_NAME).toUpperCase()
 
-  print "microserviceBuilderPipeline: registry=${registry} registrySecret=${registrySecret} build=${build} \
-  deploy=${deploy} test=${test} debug=${debug} namespace=${namespace} \
-  chartFolder=${chartFolder} manifestFolder=${manifestFolder} alwaysPullImage=${alwaysPullImage} serviceAccountName=${serviceAccountName}"
+  def maven = (config.mavenImage == null) ? 'maven:3.5.2-jdk-8' : config.mavenImage
+  def docker = (config.dockerImage == null) ? 'ibmcom/docker:17.10' : config.dockerImage
+  def kubectl = (config.kubectlImage == null) ? 'ibmcom/k8s-kubectl:v1.8.3' : config.kubectlImage
+  def helm = (config.helmImage == null) ? 'lachlanevenson/k8s-helm:v2.7.2' : config.helmImage
 
+  print "microserviceBuilderPipeline: registry=${registry} registrySecret=${registrySecret} build=${build} \
```

Comment:
Please extend this print statement to log out all the variables defined above, including `maven` `docker` `kubectl` `helm` , and any others that may have been missed.